### ISSUE TITLE
add a redirect service

### ISSUE THREAD

**Issue by only1chunts** [gigascience/gigadb-website#41]
To be done in conjunction with # 
Aim: For author that quote the private token URL in a paper instead of the actual DOI we need a way to add perminant redirects from those token pages to the correct DOI, I know of polar bear and YH fosmid papers that have done this so-far. 
and now the apple genome paper cited this token: http://gigadb.org/dataset/view/id/100189/token/BO7top5IQS1mkuyL
which should be the DOI: http://doi.org/10.5524/100189

Action: To have the option to add a redirect using the admin pages, on the "dataset" screen, a box that I can enter the URL cited by an author so that GigaDB.org will know if someone tries to go their they need to be redirected to the correct DOI. 
The "Add redirect" URL box can be added to the bottom of the admin-dataset page (see screen-shot mockup : 
![image](https://cloud.githubusercontent.com/assets/6037145/18745373/ce48423e-80b9-11e6-846e-ac6ca2191b96.png)

If a redirect is required there should be an information page displayed (for 5 seconds) to the user to explain that they have been redirected, it should look like this:
![image](https://cloud.githubusercontent.com/assets/6037145/18744398/c235f67a-80b5-11e6-9a52-0f3584a83de3.png)

**Comment by pli888** [comment]
@ScottBGI says that the private token URLs are sneaking into the PDF and online versions of the papers during the final production stage by BMC.

**Comment by only1chunts** [comment]
thats true, but since they are either unable or unwilling to make the corrections post-publication we need some way to ensure those published links end-up in the correct place.

**Comment by only1chunts** [comment]
another one just got published:, 
http://gigadb.org/dataset/view/id/100187/token/F4CDHYruxobOKmsE
should point to 
http://doi.org/10.5524/100187

**Comment by rija** [comment]
Hi, 

is it acceptable that the field from the dataset admin form for the cited url to redirect is stored in the database as a dataset attribute (e.g: associated to an  attribute of id  ‘citedurltoredirect’)? 

I feel that way is more suitable than having it in the dataset table as it's not really a core dataset property.

**Comment by only1chunts** [comment]
That is a fine idea, yes. Thanks.

**Comment by only1chunts** [comment]
@jessesiu this issue was closed because it has been done, but it doesn't seem to be working in the production server, please check why? 
The URL http://gigadb.org/dataset/view/id/100187/token/F4CDHYruxobOKmsE/ can be used as a test case. It should be added as being redirected to dataset 100187, I tried but it didn't work.

**Comment by only1chunts** [comment]
http://climb.genomics.cn/10.5524/100002 should redirect to http://dx.doi.org/10.5524/100002

**Comment by only1chunts** [comment]
This still doesn't seem to be working on the production server. We are seeing a fair number of users hitting the dataset 100114 using the token:
http://gigadb.org/dataset/view/id/100114/token/mZlMYJIF04LshpgP
So many that its appearing in the top 10 pages on Google analytics.
This needs to redirect to the DOI page:
http://dx.doi.org/10.5524/100114

**Comment by rija** [comment]
Hi @only1chunts,

Since these tokens were created for a temporary mockup page, isn't it better to delete the token and make it so that useres accessing that token based url lands in [``410 Gone``](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/410)  error page?

**Comment by only1chunts** [comment]
We've had occasions in the past where the token URL has been published so
we've needed to redirect that token URL to the correct page. That is why we
have the field in the dataset admin page for the redirect URL.
As long as that functionality works then yes the token can be deleted after
publication of a dataset.

**Comment by rija** [comment]
as a general rule, token will be removed at their end of life (when the dataset is published).
However, if a token based url has been leaked to the public, we can add static redirect in the web server configuration
for that url on ad hoc basis. Implementation of this issue is doing that for the dataset example mentioned here.

**Comment by rija** [comment]
If someone lands on a dataset through using the nginx redirect, show a message with teh same content as the old redirect notice intersticial
